sql/core/src/main/scala/org/apache/spark/sql/execution/python/RowQueue.scala

```diff
@@ -174,7 +174,8 @@ private[python] case class HybridRowQueue(
     tempDir: File,
     numFields: Int,
     serMgr: SerializerManager)
-  extends MemoryConsumer(memManager) with RowQueue {
+  extends MemoryConsumer(memManager, memManager.pageSizeBytes(), memManager.getTungstenMemoryMode)
```

Comment:
The question is which should be the default in `MemoryConsumer` constructor. Should the default be always using ON_HEAP mode? Or depending on the config?

Reply:
Default to ON_HEAP mode, I think, due to the historical reason(https://github.com/apache/spark/pull/31152#issuecomment-758703519). But it's also a little bit implicit without comments (I was considering adding comments to the constructor). Another thought is we could provide another layer interface, e.g., OnHeapMemoryConsumer for consumers who only supports ON_HEAP.

Reply:
`OnHeapMemoryConsumer` doesn't help as we still need to change every memory consumer that is on-heap only.

How about we add one more constructor to take a `MemoryManager` and `MemoryMode`? then the page size can still from the `MemoryManager`.

Reply:
Why we need that constructor? Do you mean those on-heap consumers should reuse the constructor then?

And after second thinking, I think the `MemoryConsumer` is better to require the memory mode explicitly in any cases instead of setting a default one silently. So I think we'd better remove the constructor which has the default ON_HEAP mode.

Reply:
+1 to remove that constructor. It's a bit annoying to type `memManager.pageSizeBytes()` everytime, that's why I propose to add one more constructor without the page size parameter.

Reply:
I see. I'll remove that one and add the new one.